psydac/linalg/tests/test_solvers.py

```diff
@@ -0,0 +1,492 @@
+
+import numpy as np
+import pytest
+from psydac.linalg.solvers import inverse
+from psydac.linalg.stencil import StencilVectorSpace, StencilMatrix, StencilVector
+from psydac.linalg.basic import LinearSolver
+from psydac.ddm.cart import DomainDecomposition, CartDecomposition
+
+
+def define_data_hermitien(n, p, dtype=float):
+    domain_decomposition = DomainDecomposition([n - p], [False])
+    cart = CartDecomposition(domain_decomposition, [n], [np.array([0])], [np.array([n - 1])], [p], [1])
+    # ... Vector Spaces
+    V = StencilVectorSpace(cart,dtype=dtype)
+    e = V.ends[0]
+    s = V.starts[0]
+
+    # Build banded matrix with 2p+1 diagonals: must be symmetric and positive definite
+    # Here we assign value 2*p on main diagonal and -1 on other diagonals
+    if dtype==complex:
+        factor=1+1j
+    else:
+        factor=1
+    A = StencilMatrix(V, V)
+    A[:, -p:0] = 1-1*factor
+    A[:, 0:1] = 2 * p
+    A[:, 1:p + 1] = 1-1*factor.conjugate()
+    A.remove_spurious_entries()
+
+    # Build exact solution
+    xe = StencilVector(V)
+    xe[s:e + 1] = factor*np.random.random(e + 1 - s)
+    return(V, A, xe)
+
+def define_data(n, p, matrix_data, dtype):
+    domain_decomposition = DomainDecomposition([n - p], [False])
+    cart = CartDecomposition(domain_decomposition, [n], [np.array([0])], [np.array([n - 1])], [p], [1])
+    # ... Vector Spaces
+    V = StencilVectorSpace(cart, dtype=dtype)
+    e = V.ends[0]
+    s = V.starts[0]
+
+    # Build banded matrix with 2p+1 diagonals: must be symmetric and positive definite
+    # Here we assign value 2*p on main diagonal and -1 on other diagonals
+
+    A = StencilMatrix(V, V)
+    A[:, -p:0] = -matrix_data[0]
+    A[:, 0:1] = matrix_data[1]
+    A[:, 1:p + 1] = matrix_data[2]
+    A.remove_spurious_entries()
+
+    # Build exact solution
+    xe = StencilVector(V)
+    xe[s:e + 1] = np.random.random(e + 1 - s)
+    return(V, A, xe)
+
+#===============================================================================
+@pytest.mark.parametrize( 'n', [5, 10, 13] )
+@pytest.mark.parametrize('p', [2, 3])
+@pytest.mark.parametrize('dtype', [float,complex])
+def test_bicgstab_tridiagonal( n, p, dtype ):
+
+    #---------------------------------------------------------------------------
+    # PARAMETERS
+    #---------------------------------------------------------------------------
+    if dtype==complex:
+        V, A, xe = define_data(n, p, [1-10j,6+9j,3+5j], dtype=dtype)
+    else:
+        V, A, xe = define_data(n, p, [1,6,3], dtype=dtype)
+    # Tolerance for success: L2-norm of error in solution
+    tol = 1e-10
+
+    #---------------------------------------------------------------------------
+    # TEST
+    #---------------------------------------------------------------------------
+
+    # Title
+    print()
+    print( "="*80 )
+    print( "SERIAL TEST: solve linear system A*x = b using biconjugate gradient" )
+    print( "="*80 )
+    print()
```

Comment:
Could you please remove the print statements from inside the unit test? Or even better, add the argument `verbose=False` to the test functions and only print if this is `True`.